src/responder/pam/pamsrv_passkey.c

```diff
@@ -569,6 +569,17 @@ static void pam_passkey_auth_done(int child_status,
                                   struct tevent_signal *sige,
                                   void *pvt);
 
+int pin_destructor(void *ptr)
```

Comment:
Hi,

please fix the comments about unused `ret` and incompatible types and I think you can make the destructor static as well.

As an unrelated comment, I realized that the PIN prompt `Enter PIN`  cannot be changed, I think a `pin_prompt` prompting option might be useful. But I#m fine if you prefer to add this with another PR.

bye,
Sumit

Reply:
Fixed.

Okay, I'll create a ticket for the `enter PIN` prompting option and address it separately.